polkadot/xcm/docs/src/glossary.rs

```diff
@@ -0,0 +1,88 @@
+//! # Glossary
+//!
+//! ## XCM (Cross-Consensus Messaging)
+//!
+//! A messaging format meant to communicate intentions between consensus systems.
+//! XCM could also refer to a single message.
+//!
+//! ## Instructions
+//!
+//! XCMs are composed of a sequence of instructions.
+//! Each instruction aims to convey a particular intention.
+//! There are instructions for transferring and locking assets, handling fees, calling arbitrary
+//! blobs, and more.
+//!
+//! ## Consensus system
```

Comment:
Is it a fair say a critical assumption of consensus systems is they are "asynchronous", and a lot of XCM's complexity is because it caters to this property? For example, if you want to talk between contract-a and contract-b within one chain, you would probably not use XCM.